build/templates/matchers.py.mako

```diff
@@ -0,0 +1,194 @@
+
+    def __eq__(self, other):
+        if not isinstance(other, ctypes.Array):
+            print("Unexpected type. Expected: {0}. Received: {1}".format(ctypes.Array, type(other)))
+            return False
+        if len(other) < len(self.expected_string_value) + 1:  # +1 for NULL terminating character
+            print("Unexpected length in C string. Expected at least: {0}. Received {1}".format(len(other), len(self.expected_string_value) + 1))
+            return False
+        if not isinstance(other[0], bytes):
+            print("Unexpected type. Not a string. Received: {0}".format(type(other[0])))
+            return False
+        if other.value.decode("ascii") != self.expected_string_value:
+            print("Unexpected value. Expected {0}. Received: {1}".format(self.expected_string_value, other.value.decode))
+            return False
+        return True
+
+
+# Scalars
+
+
+class ViBooleanMatcher(_ScalarMatcher):
+    def __init__(self, expected_value):
+        _ScalarMatcher.__init__(self, visatype.ViBoolean, 1 if expected_value is True else False)
+
+
+class ViSessionMatcher(_ScalarMatcher):
+    def __init__(self, expected_value):
+        _ScalarMatcher.__init__(self, visatype.ViSession, expected_value)
+
+
+class ViInt16Matcher(_ScalarMatcher):
+    def __init__(self, expected_value):
+        _ScalarMatcher.__init__(self, visatype.ViInt16, expected_value)
+
+
+class ViInt32Matcher(_ScalarMatcher):
+    def __init__(self, expected_value):
+        _ScalarMatcher.__init__(self, visatype.ViInt32, expected_value)
+
+
+class ViInt64Matcher(_ScalarMatcher):
+    def __init__(self, expected_value):
+        _ScalarMatcher.__init__(self, visatype.ViInt64, expected_value)
+
+
+class ViReal64Matcher(_ScalarMatcher):
+    def __init__(self, expected_value):
+        _ScalarMatcher.__init__(self, visatype.ViReal64, expected_value)
+
+
+# Pointers
+
+
+class ViBooleanPointerMatcher(_PointerMatcher):
+    def __init__(self):
+        _PointerMatcher.__init__(self, visatype.ViBoolean)
+
+
+class ViSessionPointerMatcher(_PointerMatcher):
+    def __init__(self):
+        _PointerMatcher.__init__(self, visatype.ViSession)
+
+
+class ViInt16PointerMatcher(_PointerMatcher):
+    def __init__(self):
+        _PointerMatcher.__init__(self, visatype.ViInt16)
+
+
+class ViInt32PointerMatcher(_PointerMatcher):
+    def __init__(self):
+        _PointerMatcher.__init__(self, visatype.ViInt32)
+
+
+class ViInt64PointerMatcher(_PointerMatcher):
+    def __init__(self):
+        _PointerMatcher.__init__(self, visatype.ViInt64)
+
+
+class ViReal64PointerMatcher(_PointerMatcher):
+    def __init__(self):
+        _PointerMatcher.__init__(self, visatype.ViReal64)
+
+
+# Buffers
+
+
+class ViBooleanBufferMatcher(_BufferMatcher):
+    def __init__(self, expected_size_or_value):
+        _BufferMatcher.__init__(self, visatype.ViBoolean, expected_size_or_value)
+
+
+class ViSessionBufferMatcher(_BufferMatcher):
```

Comment:
Do we need a ViSession buffer matcher?

Reply:
No need for it yet. I'd rather not create it now.